lib/saml_idp/request.rb

```diff
@@ -115,10 +115,11 @@ def valid_signature?
     end
 
     def service_provider?
-      service_provider.valid?
+      service_provider && service_provider.valid?
```

Comment:
I believe `ServiceProvider.valid?` is implemented correctly as-is.

The problem is this line in Request:
```
@_service_provider ||= ServiceProvider.new((service_provider_finder[issuer] || {}).merge(identifier: issuer))
```
because the resulting `@_service_provider` will always be valid even when `issuer` is nil. That is as it should be: the ServiceProvider instance *is* valid. It's the Request that is invalid.

The current fix is to not instantiate a ServiceProvider at all if there is no issuer to look up.

Since the Request can't know what the ServiceProvider class considers "valid" (respecting encapsulation) I think it is proper for Request to continue to check `valid?` before declaring it satisfactory in `service_provider?`.